PracticeGit/conversion.c

```diff
@@ -0,0 +1,41 @@
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+// conversion.c
+// Program on A3 branch.
+// ----------------------------------------------------------------------------------------------------------------------------------------------------------------
+#include "conversion.h"
+
+int binary[BITS];   // Global array to hold binary representation
+
+void DecimalToBinary (int n) {
+   for (int i = 0; i < BITS; i++)  binary[i] = (n >> i) & 1;
+}
+
+void FormatBinary (char formattedBinary[BITS / 8 * 9 + 1]) {
+   char actualBinary[BITS + 1];
+   for (int j = 0; j < BITS; j++) actualBinary[j] = binary[BITS - 1 - j] ? '1' : '0';
+   actualBinary[BITS] = '\0';
+   int pos = 0;
+   for (int j = 0; j < BITS; j++) {
+      if (j > 0 && j % 8 == 0) formattedBinary[pos++] = ' ';
+      formattedBinary[pos++] = actualBinary[j];
+   }
+   formattedBinary[pos] = '\0';
+}
+
+char ValueToHexCharacter (int value) {
+   return (value < 10) ? (char)(value + '0') : (char)(value - 10 + 'A');

```

Comment:
This function call is not required. While the statement is repeated, if is only one statement. You can just assign the value.